src/api/informer.rs

```diff
@@ -181,36 +204,9 @@ where
     }
 
     /// Watch helper
-    async fn single_watch(&self) -> Result<(Vec<WatchEvent<K>>, String)> {
+    async fn single_watch(&self) -> Result<impl Stream<Item = Result<WatchEvent<K>>>> {
         let oldver = self.version();
         let req = self.resource.watch(&self.params, &oldver)?;
-        let events = self
-            .client
-            .request_events::<WatchEvent<K>>(req)
-            .await?
-            .filter_map(|e| async move { e.ok() })
-            .collect::<Vec<_>>()
-            .await;
-
-        // Follow docs conventions and store the last resourceVersion
-        // https://kubernetes.io/docs/reference/using-api/api-concepts/#efficient-detection-of-changes
-        let newver = events
-            .iter()
-            .filter_map(|e| match e {
-                WatchEvent::Added(o) => o.meta().resourceVersion.clone(),
-                WatchEvent::Modified(o) => o.meta().resourceVersion.clone(),
-                WatchEvent::Deleted(o) => o.meta().resourceVersion.clone(),
-                _ => None,
-            })
-            .last()
-            .unwrap_or(oldver);
-        debug!(
-            "Got {} {} events, resourceVersion={}",
-            events.len(),
-            self.resource.resource,
-            newver
-        );
-
-        Ok((events, newver))
```

Comment:
so Reflector does not use single_watch anymore now?

Reply:
`Reflect` still does I think, though `Informer` doesn't anymore, I've consolidated it all into the `poll` method.